executor/src/witgen/mod.rs

```diff
@@ -300,18 +302,21 @@ impl<'a, T: FieldElement> FixedData<'a, T> {
                                 if external_values.len() != analyzed.degree() as usize {
                                     log::debug!(
                                         "External witness values for column {} were only partially provided \
-                                         (length is {} but the degree is {})",
+                                        (length is {} but the degree is {})",
                                         name,
                                         external_values.len(),
                                         analyzed.degree()
                                     );
                                 }
                             }
+                            // Remove any hint for witness columns of a later stage
```

Comment:
Why don't we filter out the whole column? Because then the IDs would not be contiguous? Maybe it makes sense to sort the witness columns by stage?

Reply:
Yeah, that's the main reason. The `ColumnMap` assumes that IDs are dense, so we'd need it to store options. Sorting by stage would help, but actually, it would also be nice if IDs would be sorted by machine, so that each machine would be responsible for a range of the IDs and could use `ColumnMap` to store machine-local rows. But it can't be sorted both by machine and stage!